From 96889273bf295cf518e511c05dc73a1c081bd6d5 Mon Sep 17 00:00:00 2001
From: josselinonduty <contact@josselinonduty.fr>
Date: Sun, 30 Nov 2025 23:59:25 +0100
Subject: [PATCH] feat: improve responsiveness on small devices

Based on patch by josselinonduty <contact@josselinonduty.fr>
Makes it possible to use the application for with a width as low as 550 pixels
without problems, and is usable even below. Should not change anything for
devices larger than 950 pixels, and nearly nothing above 800 pixels wide.
---
 build/main.js | 67 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++--
 1 file changed, 65 insertions(+), 2 deletions(-)

diff --git a/build/main.js b/build/main.js
index cbfa0c5..d801768 100644
--- a/build/main.js
+++ b/build/main.js
@@ -3572,8 +3572,8 @@
         },
         windowOptions = {
           title: "Deezer Desktop",
-          minWidth: 990,
-          minHeight: 600,
+          minWidth: 450,
+          minHeight: 450,
           icon: external_electron_namespaceObject.nativeImage.createFromPath(
             getAppIcon(external_electron_namespaceObject.app)
           ),
@@ -3710,6 +3710,69 @@
         });
     }
     loadWindow() {
+      this.appService.getWebContents().on("did-finish-load", () => {
+        const contents = this.appService.getWebContents();
+        if (!contents.getURL().startsWith(`https://${this.domain}/`)) return;
+        contents.insertCSS(`
+        #page_topbar, .page-alerts, .ads-top { min-width: 0 !important; }
+        .naboo, .page-main { min-width: auto !important; }
+
+        @media screen and (max-width: 850px) {
+          .carousel { margin-right: -24px !important; }
+        }
+        @media screen and (max-width: 808px) {
+          .container > .carousel-wrapper {
+            width: 100% !important;
+            margin: 0;
+            padding: 0;
+          }
+        }
+        @media screen and (max-width: 784px) {
+          .container { width: 100% !important; }
+        }
+        @media screen and (max-width: 950px) {
+          .page-player .player-full { min-width: auto !important; }
+          .page-player .player-queuelist .player-container {
+            margin: 32px 2% 0 !important;
+          }
+          .page-player .queuelist-cover {
+            width: 180px !important;
+            margin-right: 2% !important;
+          }
+        }
+        @media screen and (max-width: 750px) {
+          .page-player .queuelist-cover { display: none !important; }
+        }
+        @media screen and (max-width: 800px) {
+          #page_player > div:first-child { min-width: auto !important; }
+          #page_player > div:first-child > div:first-child {
+            width: calc(49% - 52px) !important;
+          }
+          #page_player > div:first-child > div:nth-child(2) {
+            padding-inline-start: unset !important;
+            padding-inline-end: unset !important;
+            width: calc(49% - 52px) !important;
+          }
+          #page_player > div:first-child > div:last-child {
+            width: auto !important;
+          }
+        }
+        @media screen and (max-width: 550px) {
+          #page_topbar .topbar-action {
+            margin-inline-start: 8px !important;
+          }
+          .channel-section > section > div:has(> .chakra-button__group) {
+            flex-wrap: wrap !important;
+          }
+          #page_player > div:first-child > div:first-child {
+            min-width: 0 !important;
+          }
+          #page_player > div:first-child > div:nth-child(2) {
+            min-width: max-content !important;
+          }
+        }
+        `, { cssOrigin: "user" }).catch(console.warn);
+      });
       const report = () => {
         const canGoBack = this.appService
             .getWebContents()
-- 
2.53.0

